cmake/modules/FindDeprecated.cmake

```
@@ -73,5 +73,12 @@ if(NOT "${Deprecated_FIND_COMPONENTS}" STREQUAL "")
                  "${Deprecated_FIND_COMPONENTS}")
 endif()
 
+if("SOURCES" IN_LIST Deprecated_FIND_COMPONENTS)
+  message(DEPRECATION
+      "Setting SOURCES prior to calling find_package() for unit tests is deprecated."
+      "Please set SOURCES after find_package and use:\n"
+      "    target_sources(testbinary PRIVATE \${SOURCES})")
```

We should only ask users to call `target_sources()`, whether or not they decide to create a `SOURCES` variable for this purpose is up to them.

How about
```suggestion
      "Setting SOURCES prior to calling find_package() for unit tests is deprecated."
      "Please add sources after find_package and use:\n"
      "    target_sources(testbinary PRIVATE <source-file.c>)")
```